dted/records/dsi.py

```diff
@@ -151,12 +152,32 @@ def from_bytes(cls, data: bytes) -> "DataSetIdentification":
             latitude_str=buffered_data.read(7).decode(_UTF8),
             longitude_str=buffered_data.read(8).decode(_UTF8),
         )
-        orientation = float(buffered_data.read(9))
-        latitude_interval = int(buffered_data.read(4)) / 10
-        longitude_interval = int(buffered_data.read(4)) / 10
-        shape = (int(buffered_data.read(4)), int(buffered_data.read(4)))[::-1]
-        coverage = float(buffered_data.read(2))
-        coverage = 1 if coverage == 0 else coverage
+        orientation = try_float(buffered_data.read(9))
+
+        latitude_interval = try_int(buffered_data.read(4))
+        if latitude_interval is None:
+            raise InvalidFileError(
+                "The latitude interval of the gridded data must be specified in the "
+                "DataSetIdentification section of the DTED file. "
+            )
+
+        longitude_interval = try_int(buffered_data.read(4))
+        if longitude_interval is None:
+            raise InvalidFileError(
+                "The longitude interval of the gridded data must be specified in the "
+                "DataSetIdentification section of the DTED file. "
+            )
+
+        shape = (try_int(buffered_data.read(4)), try_int(buffered_data.read(4)))[::-1]
+        if shape[0] is None or shape[1] is None:
+            raise InvalidFileError(
+                "The shape of the gridded data must be specified in the "
+                "DataSetIdentification section of the DTED file. "
+            )
+
+        coverage = try_float(buffered_data.read(2))
+        if coverage is not None:
```

Comment:
The if doesn't look like it's needed here. If try_float returns None, the next line, "coverage=1 if coverage == 0 else coverage" would still keep coverage at None

Reply:
good catch!